modal/_asgi.py

```diff
@@ -50,12 +51,27 @@ def wsgi_app_wrapper(wsgi_app):
     return asgi_app_wrapper(asgi_app)
 
 
-def webhook_asgi_app(fn: Callable, method: str):
+def _stream_generator(fn: Callable) -> Callable:
+    # Wrap a generator function in a handler that returns a StreamingResponse
+    from fastapi.responses import StreamingResponse
+
+    @functools.wraps(fn)
+    async def wrapper(*args, **kwargs):
+        generator = fn(*args, **kwargs)
+        return StreamingResponse(generator)
```

Comment:
Need to set a `media_type` here otherwise webhook returns `content-type: None` and streaming response won't work in browsers at least (I think it still works with `curl --no-buffer`). 

Setting `media_type="text/html"` on behalf of the user is better than nothing, and gets the browser case to work, but ideally there'd be an easy way for users to set their content-type.

Reply:
Maybe it's better to not do the magic thing here then? We could catch it if the user function is a generator and tell them to use StreamingResponse. or wdyt?

Reply:
I'm ok with defaulting to text/html though. I guess that's what fastapi does if you just return a byte stream anyway. Just wondering if the case for streaming html is limited anyway. If 99% of the time involves some other content type then maybe it's better to force the user to be explicit. Not sure.

Reply:
Actually it isn't `content-type: none` with this snippet, it's JSON. Still not a streaming response though. 

```python
@stub.webhook()
def app():
    import time
    for i in range(10):
        yield f"frame {i}: hello world!".encode()
        time.sleep(1.0)
```

Reply:
Yeh less magic sounds good. Catch and offer a suggestion to return a `StreamingResponse` with appropriate `media_type` set. 

Currently the async test case snippet doesn't work if I run it locally with `MODAL_SYNC_ENTRYPOINT`. I get this error: 

```
ValueError: [TypeError("'async_generator' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
```